### PR TITLE
DDPM changes to support v diffusion

### PR DESCRIPTION
**Pull request by bglick13**
Trying to keep this a small PR that just adds necessary changes in DDPM scheduler to support v diffusion. I have a supporting example script using the butterflies dataset I can add either here or as a separate PR

### CODE REVIEW

File: src/diffusers/pipelines/ddpm/pipeline_ddpm.py [hunk 1]
```diff
@@ -44,6 +44,7 @@ def __call__(
         generator: Optional[torch.Generator] = None,
         output_type: Optional[str] = "pil",
         return_dict: bool = True,
+        prediction_type: Literal["epsilon", "sample", "v"] = "epsilon",
```
[the remark this answers is not in the document]
Comment (patrickvonplaten) [reply src/diffusers/pipelines/ddpm/pipeline_ddpm.py:47]: Instead you could use `self.scheduler.config.variance_type` further below

File: src/diffusers/schedulers/scheduling_ddpm.py [hunk 2]
```diff
@@ -221,7 +234,7 @@ def step(
         model_output: torch.FloatTensor,
         timestep: int,
         sample: torch.FloatTensor,
-        prediction_type: str = "epsilon",
```
Comment (patrickvonplaten) [inline src/diffusers/schedulers/scheduling_ddpm.py:224]: Actually I think we should deprecate this argument and just have it in the config instead - I don't it's something that differs for every scheduler call. @anton-l what do you think?

File: src/diffusers/pipelines/ddpm/pipeline_ddpm.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (bglick13) [reply src/diffusers/pipelines/ddpm/pipeline_ddpm.py:47]: That makes sense! That means the pipeline doesn’t have to change at all either

File: src/diffusers/schedulers/scheduling_ddpm.py (hunk 2 above)
Comment (bglick13) [reply src/diffusers/schedulers/scheduling_ddpm.py:224]: That makes sense! That means the pipeline doesn’t have to change at all either

Comment (anton-l) [reply src/diffusers/schedulers/scheduling_ddpm.py:224]: I agree, the prediction type doesn't need to change during sampling

**Comment by bglick13** [comment]
By the way, here is the butterflies example samples trained on this implementation. I think it looks good compared to the original example, but wanted to put it out there for others to evaluate.

**Pull request merged**